js/ui/html_editor/ui/formDialog.js

```diff
@@ -115,6 +111,19 @@ class FormDialog {
         }, this._popupUserConfig);
     }
 
+    onHiding() {
+        this.beforeAddButtonAction = undefined;
+        this.deferred.reject();
+    }
+
+    callAddButtonAction(event) {
+        if(this.beforeAddButtonAction && !this.beforeAddButtonAction()) {
+            return;
+        }
```

Comment:
Minor comment.

Let's make this if condition shorter:
```suggestion
        if(!this.beforeAddButtonAction?.()) {
            return;
        }
```